chapter2/tests/t6/chapter2-l2-t6-get-file-unique-dirs.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+#
+# The script return unique directories of all files (including nested directories).
+set -eou pipefail
+
+IFS=$'\n'
+
+dir="${1}"
+
+if [[ ! -e "${dir}" || ! -d "${dir}" ]]; then
+  echo "The '${dir}' does not exist or it is not a directory." >&2
+  exit 1
+fi
+
+#######################################
+# Get files directories (including nested directories).
+# Arguments:
+#   Directory.
+# Outputs:
+#   Writes directories into stdout.
+#######################################
+function get_files_directories {
+  local dir="${1}"
```

Comment:
No need to repeat again `dir` variable, you already have it as a global var and can use it in function without `local` definition